api/agentpb/agent.go

```diff
@@ -86,27 +84,26 @@ func GetAgentConnectMetadata(ctx context.Context) AgentConnectMetadata {
 
 // SendAgentServerMetadata sends metadata to pmm-agent.
 // Used by pmm-managed.
-func SendAgentServerMetadata(stream grpc.ServerStream, md AgentServerMetadata) error {
+func SendAgentServerMetadata(stream grpc.ServerStream, md *AgentServerMetadata) error {
 	header := metadata.Pairs(
 		mdAgentNodeID, md.AgentRunsOnNodeID,
 		mdServerVersion, md.ServerVersion,
 	)
 	if err := stream.SendHeader(header); err != nil {
-		return status.Errorf(codes.Internal, "Can't send server metadata to client.")
+		return err
 	}
-
 	return nil
 }
 
 // GetAgentServerMetadata receives metadata from pmm-managed.
 // Used by pmm-agent.
-func GetAgentServerMetadata(stream grpc.ClientStream) (AgentServerMetadata, error) {
-	var res AgentServerMetadata
+func GetAgentServerMetadata(stream grpc.ClientStream) (*AgentServerMetadata, error) {
 	md, err := stream.Header()
 	if err != nil {
-		return res, err
+		return nil, err
 	}
 
+	res := &AgentServerMetadata{}
```

Comment:
```suggestion
	var res AgentServerMetadata
```

https://dave.cheney.net/practical-go/presentations/qcon-china.html#_use_a_consistent_declaration_style